iam.yml

```diff
@@ -0,0 +1,120 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  Define IAM Roles utilized to build, deploy, and operate the JavaBuilder service.
+  Note: CAPABILITY_IAM capability is required to deploy this stack.
+
+Parameters:
+  ArtifactBucket:
+    Type: String
+    Description: Name of existing S3 bucket for storing pipeline artifacts
+  TemplateBucket:
+    Type: String
+    Description: Name of existing S3 bucket for storing CloudFormation template artifacts
+
+Resources:
+  JavaBuilderLambdaExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal: {Service: [lambda.amazonaws.com]}
+      Path: /
+      Policies:
+        - PolicyName: JavaBuilderLambdaExecutionPolicy
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "codepipeline:GetPipelineState"
+                  - "codepipeline:GetPipelineExecution"
+                  - "codepipeline:PutApprovalResult"
+                  - "cloudformation:DescribeStacks"
+                Resource: '*'
+              - Effect: Allow
+                Action:
+                  - "xray:PutTraceSegments"
+                  - "xray:PutTelemetryRecords"
+                Resource: '*'
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+  JavaBuilderCloudFormationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal: {Service: [cloudformation.amazonaws.com]}
+        Version: '2012-10-17'
+      Path: /
+      Policies:
+        - PolicyName: ManageResourcesAccess
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "apigateway:*"
+                  - "lambda:*"
+                  - "cloudformation:*"
+                  - "route53:*"
```

Comment:
I think we'll need something about SQS here as well.

Reply:
If it's helpful, a superset of the permissions needed for the API Gateway lambda is here: https://console.aws.amazon.com/iam/home?#/roles/javaBuilderOnConnect-role-3phgtweq?section=permissions
And a superset of the permissions needed for the long-running lambda is here: https://console.aws.amazon.com/iam/home#/roles/javaBuilderExecuteCode-role-4ugmok8v?section=permissions